### PR TITLE
WIP : Adds Painkiller Addictions

### PR DESCRIPTION
**Pull request by Diegoflores31**
# About the pull request

THIS IS A SHOWCASE . THE CHANGELOG AND FILES STILL NEED POLISH

Adds Opiod addiction , chemicals that are considered opioids will now slowly buildup addiction. each chemical has a different addiction factor , the stronger the chemical the easier it will hook you up.

**How it works ?**
each tick you have an oppioid on you system you build up addiction . each tick clean contribute will naturally lower your addiction level. each chemical has a different level .

after being dosed with oppioids for X amount of time winout being clean your addiction stage will increase.
while you have oppioids on you system regardless of the stage you will suffer no symptoms from addiction.
TODO: you can dose yourself with sleen to keep the addiction "in control"

being clean for 4 minutes will revert 1 stage of addiction.

stage 1 unnoticeable harmless
stage 2 shows cravings but otherwise harmless
stage 3 starts to experiment shaking , urgent need for dosing , headaches , etc
stage 4 hallucinations , chemical resistance (opioid trait) , stronger effects 

Consequences to being addicted include : 
- headaches
- shaking 
- flavor messages 
- at the final stage of addiction you will receive the "Opiod receptor" trait that makes all painkillers less effective on you untill you seek medical help to fix it or stay dosed .(shipside)

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Adds Opioid addiction "virus"
balance: Opioid painkillers can generate addiction (tramadol , oxycodone , Sleen)
/:cl:

### CODE REVIEW

**Comment by Korey45** [comment]
This would very highly value a section in "Why its good for the game". It's going to be absolutely dogpiled and hated without a good enough reason, Showcase or not.

File: code/datums/diseases/addiction.dm [hunk 1]
```diff
@@ -128,3 +128,91 @@
...
+
+		if(2) // Regular Use
+			if(prob(5))
+				var/message = SPAN_INFO( pick("One more wont hurt; everything is under control.", "I will quit when things settle down, but right now I just need this.", "I have been through worse. I can handle this for now.", "I am CURED ; I am ready to use painkillers again" ) )
+				to_chat(affected_mob, message)
+				return
+
+		if(3) // Dependance
+			if(prob(5))
+				var/message = SPAN_DANGER( pick("I am not strong enough , I NEED IT", "If I dont get some soon, I dont know what is going to happen.", "I cant think straight. I need it", "I cant handle this pain anymore, I need it now." ) )
```
Comment (vero5123) [inline code/datums/diseases/addiction.dm:180]:
```suggestion
				var/message = SPAN_DANGER( pick("I am not strong enough, I NEED IT!", "If I don't get some soon, I don't know what is going to happen.", "I cant think straight. I need it.", "I cant handle this pain anymore, I need it now." ) )
```

File: code/datums/diseases/addiction.dm [hunk 2]
```diff
@@ -128,3 +128,91 @@
...
+				return
+
+		if(4) // Tolerance
+			if(!CHECK_BITFIELD(affected_mob.disabilities, OPIATE_RECEPTOR_DEFICIENCY))
+				ENABLE_BITFIELD(affected_mob.disabilities, OPIATE_RECEPTOR_DEFICIENCY)
+				to_chat(affected_mob, SPAN_HIGHDANGER("Your stomach drops down . something terrible has happened"))
+				return
+
+			if(prob(5))
+				var/message = SPAN_DANGER( pick("I NEED IT NOW", "If I dont get some soon, I dont know what is going to happen.", "I cant think straight. I need it", "I cant handle the pain anymore." ) )
```
Comment (vero5123) [inline code/datums/diseases/addiction.dm:201]:
```suggestion
				var/message = SPAN_DANGER( pick("I NEED IT NOW!", "If I don't get some soon, I don't know what is going to happen.", "I cant think straight. I need it", "I cant handle the pain anymore." ) )
```

File: code/modules/reagents/chemistry_properties/prop_negative.dm [hunk 3]
```diff
@@ -451,12 +451,30 @@
 	holder.custom_metabolism = holder.custom_metabolism * (1 + POTENCY_MULTIPLIER_VLOW * level)
 	..()
 
+/datum/chem_property/negative/opioid
+	name = PROPERTY_OPIOID
+	code = "OPD"
+	description = "This chemical contains opiate traces , making it highly addictive"
```
Comment (vero5123) [inline code/modules/reagents/chemistry_properties/prop_negative.dm:457]:
```suggestion
	description = "This chemical contains opiate traces, making it highly addictive."
```

File: code/modules/reagents/chemistry_reagents/other.dm [hunk 4]
```diff
@@ -162,13 +162,14 @@
 /datum/reagent/sleen
 	name = "Sleen"
 	id = "sleen"
-	description = " A favorite of marine medics, it is an illicit mixture of name brand lime soda and oxycodone, known for it's distinct red hue. Overdosing can cause hallucinations, loss of coordination, seizures, brain damage, respiratory failure, and death."
+	description = " A favorite of marine medics, it is an illicit mixture of name brand lime soda and oxycodone, known for it's distinct green color.The chemical flavoring increase the painkilling power , usually used by junkies as a replacement of the real deal"
```
Comment (vero5123) [inline code/modules/reagents/chemistry_reagents/other.dm:165]:
```suggestion
	description = " A favorite of marine medics, it is an illicit mixture of name brand lime soda and oxycodone, known for its distinct green color. The chemical flavoring increase the painkilling power, usually used by junkies as a replacement of the real deal."
```

**Review by vero5123 — COMMENTED** [review]
Very cool pr, hopefully it gets merged.

File: code/datums/diseases/addiction.dm [hunk 5]
```diff
@@ -128,3 +128,91 @@
...
+				to_chat(affected_mob, message)
+				return
+
+			if(prob(1))
+				to_chat(affected_mob, SPAN_HIGHDANGER("RUN RUN RUN RUN RUN RUN RUN RUN RUN RUN RUN RUN"))
+				playsound_client(affected_mob.client, 'sound/effects/heart_beat_short.ogg', 35)
+				affected_mob.apply_effect(3, DAZE)
+				affected_mob.make_jittery(150)
+				affected_mob.emote("scream")
+				return
```
Comment (vero5123) [inline code/datums/diseases/addiction.dm:211]: 1GrrAjNOAE-1

**Comment by SASoperative** [comment]
How exactly would you seek medical help shipside for this? Because if it is just a basic medical antidote like normal SS13 viruses, and not some form of surgery, or machine needed, than what is to stop the cure from being deployed planetside or with combat medics? Not like I would like that because playing a combat medic can already be hell as is in its current state. 

I also do not see any additional notes about the vendor changes removing Oxycodone bottles, or the changing of tramadol from basic medical kits to paracetamol, or the variable changes that make painkillers overall weaker. 

I know that you attempted to make a PR for another change earlier, did you forget to split or revert those changes before starting this one? 
I say this because the commits are from yesterday, but the PR for this addiction change was made today.

**Comment by Diegoflores31** [comment]
> If you cant be bothered to justify the merits of your changes, the community will not understand your changes.
> 
> Please dont be lazy, and actually fill out the PR template?

as on the main tittle i am still kinda finishing up the PR . i just uploaded it because i wanted to show it to other contribs 
this is just a showcase of the code untill i finish it off

**Comment by Diegoflores31** [comment]
> I know that you attempted to make a PR for another change earlier, did you forget to split or revert those changes before starting this one? I say this because the commits are from yesterday, but the PR for this addiction change was made today.

they are files from another PR that got mixed up . they are being deleted , its written on the description

image

**Comment by SASoperative** [comment]
> > I know that you attempted to make a PR for another change earlier, did you forget to split or revert those changes before starting this one? I say this because the commits are from yesterday, but the PR for this addiction change was made today.
> 
> they are files from another PR that got mixed up . they are being deleted , its written on the description
> 
> image

Understood, I was uncertain on what that exactly meant by files needing polish implied. 
My apologies.

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
image

this section exists for a reason, fill it out before marking it as "ready"; if it's not ready, that's what drafting's for

**Comment by Derunner** [comment]
Medicord wants to know your exact location

**Comment by Diegoflores31** [comment]
> this section exists for a reason, fill it out before marking it as "ready"; if it's not ready, that's what drafting's for

i wanted to but the draft button didnt appear on my screen

**Comment by efzapa** [comment]
far better than flat organ damage, very based

**Comment by TheManWithNoHands** [comment]
Very cool. But SAS is rigth, the additcion needs a cure that is avalibel only with proper medical treatment, and not just more drugs.
If marines can just slip their no addiction juice in the unga bottle, that makes it kinda useless.

(Custom research chems excluded, also make sure the anti addicting property works for this btw. Sounds like a buff to stims, but it aint. Cause that means you would have to decide between stim without painkilling, or using up two property slots. Wich is a fun choice.)

Legit, if you want some ideas, look at TGMC medical system. They tried to reduce blind premedding like you currently are. They reworked their medical system a LOT for that.
Good idea to see what solutions they came up with. Migth be worth yoinking some of their ideas. Like, they made changes to what organ dmg causes, to allow chems to cause organ dmg while keeping it interesting. For exmp heart dmg lowers your robustness, kenshi style. Aka, you go pain crit faster. Or kindey/liver (i forgot wich one) dmg decreases how well chems work.
Def worth a peek.

**Comment by Katskan** [comment]
I'd love to have a conversation with you about IRL pain medications that would do a great job of offering new options and depth to gameplay as well as have their own up and downsides, without disrupting gameplay too terribly much. 

An example is the distinction between narcotic and non-narcotic painkillers. Narcs could be made to cause respiratory depression in the form of oxygen damage which is manageable with dexalin for lower strength meds and needs dex+ for higher strength surgical meds. Non-narcotics have other effects, such as toradol which has an anti-platelet effect, resulting in faster bleeding, or ketamine with dissociative and hallucinogenic effects. There's a lot of options to expand with that have built-in pros and cons without using the addiction system.

File: code/modules/reagents/chemistry_reagents/other.dm [hunk 6]
```diff
@@ -162,13 +162,14 @@
 /datum/reagent/sleen
 	name = "Sleen"
 	id = "sleen"
-	description = " A favorite of marine medics, it is an illicit mixture of name brand lime soda and oxycodone, known for it's distinct red hue. Overdosing can cause hallucinations, loss of coordination, seizures, brain damage, respiratory failure, and death."
+	description = " A favorite of marine medics, it is an illicit mixture of name brand lime soda and oxycodone, known for its distinct green color. The chemical flavoring increase the painkilling power, usually used by junkies as a replacement of the real deal."
```
Comment (vero5123) [inline code/modules/reagents/chemistry_reagents/other.dm:165]:
```suggestion
	description = "A favorite of marine medics, it is an illicit mixture of name brand lime soda and oxycodone, known for its distinct green color. The chemical flavoring increase the painkilling power, usually used by junkies as a replacement of the real deal."
```

File: code/datums/diseases/addiction.dm [hunk 7]
```diff
@@ -128,3 +128,91 @@
...
+
+		if(4) // Tolerance
+			if(!CHECK_BITFIELD(affected_mob.disabilities, OPIATE_RECEPTOR_DEFICIENCY))
+				ENABLE_BITFIELD(affected_mob.disabilities, OPIATE_RECEPTOR_DEFICIENCY)
+				to_chat(affected_mob, SPAN_HIGHDANGER("You feel your stomach drop, something terrible has happened."))
+				return
+
+			if(prob(5))
+				var/message = SPAN_DANGER( pick("I NEED IT NOW!", "If I don't get some soon, I don't know what is going to happen.", "I cant think straight. I need it", "I cant handle the pain anymore." ) )
```
Comment (vero5123) [inline code/datums/diseases/addiction.dm:201]:
```suggestion
				var/message = SPAN_DANGER( pick("I NEED IT NOW!", "If I don't get some soon, I don't know what is going to happen.", "I cant think straight. I need it.", "I cant handle the pain anymore." ) )
```

**Comment by Steelpoint** [comment]
You know, under "realistic" circumstances, opioid addiction is a very long-term issue to resolve. People who are have abused opioids can reach a point where conventional painkillers are nearly useless for genuine pain-relief, in some cases you are left with someone who literally can not be relieved of pain even if given some of the strongest medications available, which sucks if you also have chronic (or very acute) pain. 

At that point you might need medications like methadone or buprenorphine to be used as treatment options for opioid addition, which are also fairly strong painkillers in their own right. 

Of course, at this point we may as well make it that Doctors need to write a prescription for Tramadol or Oxycodone and you need to get two Nurses to sign out every single tablet of said medications from a special vendor before they can give 1 pill to a Marine. I look forward to the Provost fax regarding a miscount of the Oxycodone medication. 

All this is saying, is this level of complication in regards to painkiller addiction really necessary for Colonial Marines? A (relativity) fast paced combat game. If half the Marine force is suddenly unable to use painkillers, that might have a disruptive effect on the game. 

Perhaps if this was a optional penalty you could take as a Rifleman or shipside roleplay role, but something forced on everyone? I'm not so sure.

**Comment by TheManWithNoHands** [comment]
> You know, under "realistic" circumstances, opioid addiction is a very long-term issue to resolve. People who are have abused opioids can reach a point where conventional painkillers are nearly useless for genuine pain-relief, in some cases you are left with someone who literally can not be relieved of pain even if given some of the strongest medications available, which sucks if you also have chronic (or very acute) pain.

My guy, its a punishment for exsessive premedding. That the punishment is called opioid addiction is IC flavor.
Dont start pulling out those arguments. You been dev long enough to know how silly those are.

**Comment by Korey45** [comment]
Pre-medicating isn't even an issue in the first place. I don't know why that's a valid reason. Its not been explained at all why its an issue.

**Comment by frozentsbgg** [comment]
We have like six different levels of painkillers holding over from a system that was removed.
I'm all for attempts to bring back addiction. 
There should be a functional difference between Tramadol and oxytocin from a painkiller and downside perspective.

**Comment by TheManWithNoHands** [comment]
> Pre-medicating isn't even an issue in the first place. I don't know why that's a valid reason. Its not been explained at all why its an issue.

Looks more like an attempt to make the medical system more interesting then a respons to a direct problem. We have a pretty simple med system after all.

**Comment by Venuska1117** [comment]
**_We are space marines, you overdosed marine?! 
GET YOUR ASS BACK TO SHIP AND GO TO AUTOSURGERY MACHINE FOR OD TRETAMENT!_**

_*cough*_  my idea to solve this issue is to give autosurgery machine that i forgot name, ability to cure OD's, because if you make chemical, people will just put it in their "Own Mix of chems" and it will remove need to go shipside for OD healing, but thats my idea. (there are some groundsite too)

**Comment by TheManWithNoHands** [comment]
> __cough__ my idea to solve this issue is to give autosurgery machine that i forgot name

You mean the dialysis mashine?
That already cures OD.
Yb removing all chems.

**Comment by Nanu308** [comment]
Painkiller addiction was attempted in the far past and was really bad for Marines and a directly unfun gameplay mechanic to deal with. Going to place DNM on it due to that.

**Comment by harryob** [comment]
pr is blocked and untouched for two weeks, feel free to ask us to reopen/make a new pr if you work it out with nanu

**Pull request closed without merging**